orc8r/gateway/go/Makefile

```diff
@@ -19,6 +19,17 @@ endif
 GOOS ?= $(shell go env GOOS)
 GOARCH ?= $(shell go env GOARCH)
 
+MAGMA_BUILD_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD)
+MAGMA_BUILD_TAG ?= $(shell git describe --tags --abbrev=0)
+MAGMA_BUILD_COMMIT_HASH ?= $(shell git rev-parse HEAD)
+MAGMA_BUILD_COMMIT_DATE ?= $(shell git log -1 --format=%cd)
```

Comment:
The reason I added build.py and a separate script `get_version_info.sh` on my other PR is to be able to have the version in the image itself. But wit this solution that may not be necessary anymore

But if we want images to have envars, then we need to use `get_version_info.sh` 

https://github.com/magma/magma/pull/7836/files#diff-6c789c1e289eb4ec3c48a43f7366a38658b5acf706f0ea81884c822977634490R1

Reply:
This is just a makefile for Gateway's magmad & some other optional services.
magmad is not always built under docker or on VMs (see https://github.com/magma/magma/blob/master/openwrt/gateway/deploy/build.sh#L183 for example)

You may still want to use something like get_version_info.sh for other modules (FeG) instead or in addition to this style of makefile changes.
'?=' type of assignment means that if there is already env variable set, it won't be overwritten, but if the variable is not defined - it'll be set.